### PR TITLE
enable alert_manager

### PR DESCRIPTION
**Pull request by Emterry**
enabling alert_manager to connect to pagerduty services

### CODE REVIEW

File: terraform/pagerduty/modules/service/technical/main.tf [hunk 1]
```diff
@@ -275,3 +275,34 @@ resource "aws_secretsmanager_secret_version" "pagerduty_airflow_integration_key"
   secret_id     = aws_secretsmanager_secret.pagerduty_airflow_integration_key[0].id
   secret_string = pagerduty_service_integration.airflow[0].integration_key
 }
+
+##################################################
+# Alert Manager Integration
+##################################################
+
+data "pagerduty_vendor" "alert_manager" {
+  name = "Prometheus"
```
Comment (julialawrence) [inline terraform/pagerduty/modules/service/technical/main.tf:284]: Should this be `Alert Manager` or do you expect we'll need a separate integration for each source that feeds alerts into the manager?

File: terraform/pagerduty/modules/service/technical/main.tf [hunk 2]
```diff
@@ -275,3 +275,34 @@ resource "aws_secretsmanager_secret_version" "pagerduty_airflow_integration_key"
...
+}
+
+resource "pagerduty_service_integration" "alert_manager" {
+  count = var.enable_alert_manager_integration ? 1 : 0
+
+  name    = data.pagerduty_vendor.alert_manager.name
+  service = pagerduty_service.this.id
+  #vendor = data.pagerduty_vendor.alert_manager.id
```
Comment (julialawrence) [inline terraform/pagerduty/modules/service/technical/main.tf:292]: Could you please annotate why this is currently commented out? We will forget. :)

**Review by julialawrence — CHANGES_REQUESTED** [review]
Looks good. Two small questions.

File: terraform/pagerduty/modules/service/technical/main.tf (hunk 1 above)
Comment (Emterry) [reply terraform/pagerduty/modules/service/technical/main.tf:284]: when i ran the vendor.sh for pagerduty it doesnt provide alertmanager as a vendor - it only provides the vendor id for prometheus alert manager ( the name it required was just prometheus)

File: vendor.sh
Comment (jacobwoffenden) [inline vendor.sh:1]: move to `scripts/pagerduty/list-vendors.sh`

File: terraform/pagerduty/modules/service/technical/main.tf [hunk 3]
```diff
@@ -275,3 +275,34 @@ resource "aws_secretsmanager_secret_version" "pagerduty_airflow_integration_key"
...
+  vendor = "P1S0PW6"
+}
+
+resource "aws_secretsmanager_secret" "pagerduty_alert_manager_integration_key" {
+  count = var.enable_alert_manager_integration ? 1 : 0
+
+  name       = "${local.secretsmanager_prefix}/alert_manager"
```
Comment (jacobwoffenden) [inline terraform/pagerduty/modules/service/technical/main.tf]:
```suggestion
  name       = "${local.secretsmanager_prefix}/alert-manager"
```

File: terraform/pagerduty/technical-services.tf
Comment (jacobwoffenden) [inline terraform/pagerduty/technical-services.tf:1]: I would enable it on the compute service, not security

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]

**Review by julialawrence — DISMISSED** [review]

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]
It's still enabled on Analytical Platform Security service

**Review by jacobwoffenden — APPROVED** [review]

**Pull request merged**